gym/tf_od_api/workspace/train_core/model_main.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Binary to run train and evaluation on object detection model."""
+
+from __future__ import absolute_import
```

Comment:
If we are using python3, we should probably remove these future statements... They are usually useful for backwards compatibility, but we probably want to stick with a bunch of specific versions so we are sure we have exactly the same environment across everything. We can leverage miniconda when defining the docker file. For example, your Dockerfile right now doesn't have any specification on which python version it's using (ie it just uses "python").


In my opinion, you should generate a .yaml file from your current anaconda environment:
https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#exporting-the-environment-yml-file

And then, when you create the anaconda environment in the docker you specify that .yaml file:
https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-from-an-environment-yml-file

This will also save us in the case of tf being uploaded and breaking compatibility: instead of installing tensorflow-gpu, you specify which version you really want.



I'm guessing you can also use miniconda rather than anaconda for the docker, to have a lighter one:
https://hub.docker.com/r/continuumio/miniconda

Reply:
We won't need that module any longer, I will remove it

Reply:
Ok, cool. Specifications of module dependencies using a .yml file is probably important already for this PR as well. I think it helps in having the deployment more organised.

Reply:
for now I am using Dockerfile, when the pipeline is more stable, then maybe can move to .yml